src/__tests__/LoginScreen.test.tsx

```
@@ -0,0 +1,594 @@
+      consoleSpy.mockRestore();
+    });
+
+    test('logs empty password attempts', async () => {
+      const consoleSpy = jest.spyOn(console, 'log').mockImplementation();
+      const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+      renderWithRouter(<LoginScreen />);
+      
+      const passwordInput = screen.getByLabelText(/password/i);
+      
+      // Somehow trigger form submission with empty password (edge case)
+      fireEvent.submit(passwordInput.closest('form')!);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)